CMakeLists.txt

```diff
@@ -1,284 +1,41 @@
-set(CMAKE_CXX_FLAGS_GCSTATS "${CMAKE_CXX_FLAGS_DEBUG}")
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${K_LLVM_BACKEND_LTO_FLAG}")
-set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} ${K_LLVM_BACKEND_LTO_FLAG} -fno-omit-frame-pointer")
-set(CMAKE_EXE_LINKER_FLAGS_FASTBUILD "${CMAKE_EXE_LINKER_FLAGS_RELEASE}")
-set(CMAKE_EXE_LINKER_FLAGS_GCSTATS "${CMAKE_EXE_LINKER_FLAGS_DEBUG}")
-set(CMAKE_EXE_LINKER_FLAGS_RELEASE "${CMAKE_EXE_LINKER_FLAGS_RELEASE} ${K_LLVM_BACKEND_LTO_FLAG}")
-set(CMAKE_EXE_LINKER_FLAGS_RELWITHDEBINFO "${CMAKE_EXE_LINKER_FLAGS_RELEASE} ${K_LLVM_BACKEND_LTO_FLAG}")
-
-if(APPLE)
-  if(NOT USE_NIX)
-    if(DEFINED ENV{HOMEBREW_PREFIX})
-      set(BREW_PREFIX $ENV{HOMEBREW_PREFIX})
-    else()
-      execute_process(
-        COMMAND brew --prefix
-        OUTPUT_VARIABLE BREW_PREFIX
-        ERROR_VARIABLE BREW_ERROR
-        RESULT_VARIABLE BREW_RESULT
-        OUTPUT_STRIP_TRAILING_WHITESPACE)
-
-      if(NOT BREW_RESULT EQUAL 0)
-        message(WARNING "Error running brew --prefix; you may need to manually configure package search paths.")
-        message(WARNING "  : ${BREW_ERROR}")
-      endif() # BREW_RESULT
-    endif() # ENV{HOMEBREW_PREFIX}
-
-    message(STATUS "Looking for Homebrew dependencies in ${BREW_PREFIX}")
-    include_directories(AFTER SYSTEM "${BREW_PREFIX}/include")
-    link_directories(AFTER "${BREW_PREFIX}/lib")
-    set(ENV{PKG_CONFIG_PATH} "${BREW_PREFIX}/opt/libffi/lib/pkgconfig")
-  endif() # USE_NIX
-endif() # APPLE
-
-find_package(PkgConfig REQUIRED)
-pkg_check_modules(FFI REQUIRED libffi)
-if(USE_NIX)
-include_directories(AFTER ${FFI_INCLUDE_DIRS})
-link_directories(AFTER ${FFI_LIBRARY_DIRS})
-else() # USE_NIX
-include_directories(AFTER SYSTEM ${FFI_INCLUDE_DIRS})
-link_directories(AFTER ${FFI_LIBRARY_DIRS})
-endif() # USE_NIX
-
-if ((CMAKE_BUILD_TYPE STREQUAL "Release" OR CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
-     AND K_LLVM_BACKEND_LTO)
-set(LLVM_KOMPILE_LTO "lto")
-else()
-set(LLVM_KOMPILE_LTO "nolto")
-endif()
-
-set(CMAKE_AR "${CMAKE_C_COMPILER_AR}")
-set(CMAKE_RANLIB "${CMAKE_C_COMPILER_RANLIB}")
-
-find_package(LLVM 15 QUIET CONFIG)
 
-if (NOT LLVM_FOUND)
-  find_package(LLVM 14 QUIET CONFIG)
-endif()
-
-if (NOT LLVM_FOUND)
-  find_package(LLVM 13 QUIET CONFIG)
-endif()
-
-if (NOT LLVM_FOUND)
-  find_package(LLVM 12 QUIET CONFIG)
-endif()
+include(CTest)
 
-if (NOT LLVM_FOUND)
-  find_package(LLVM 11.1 QUIET CONFIG)
-endif()
+# Needs to be included first so that pkg-config paths are set up when looking
+# for dependencies.
+include(FixHomebrew)
 
-if (NOT LLVM_FOUND)
-  find_package(LLVM 11 REQUIRED CONFIG)
-endif()
+add_subdirectory(deps)
+include(FindDependencies)
 
-message(STATUS "Found LLVM ${LLVM_PACKAGE_VERSION}")
-if (${LLVM_PACKAGE_VERSION} VERSION_LESS 11)
-  message(FATAL_ERROR "LLVM 11 or newer is required")
-endif()
+include(KLLVMCompilerFlags)
 
 include(SetTriple)
 
-set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
```

Comment:
Removed this default setting as a stylistic choice; tends to be used for building downstream tooling or debugging a build system.